### PR TITLE
chore(api,robot-server): add rpc and session logging.

### PR DESCRIPTION
**Pull request by amitlissack**
# Overview

There are several reported bugs related to pause/resume. This PR adds some logging to show when user is trying to pause and resume.

# Changelog

Logging

# Review requests

# Risk assessment

None. Just logging.

### CODE REVIEW

File: api/src/opentrons/api/session.py [hunk 1]
```diff
@@ -586,6 +586,8 @@ def set_state(self, state: 'State',
                   reason: str = None,
                   user_message: str = None,
                   duration: float = None) -> None:
+        log.info(f"Session.set_state state={state} "
+                 f"user_message={user_message} duration={duration}")
```
Comment (SyntaxColoring) [inline api/src/opentrons/api/session.py]: Should we have `reason` in here for completeness?

I'm surprised the Python standard library doesn't include a decorator for logging a function call and all its arguments. Maybe it's worth doing our own.

**Review by SyntaxColoring — APPROVED** [review]
I think this will be helpful.

Optionally, if we wanted to go further, we could add logging on the higher-level functions like `pause()`. That could help identify deadlocks inside the higher-level functions before they call `self.set_state()`.

**Review by sfoster1 — APPROVED** [review]
This looks good, I agree with @SyntaxColoring that it should log the `reason`.

File: api/src/opentrons/api/session.py (hunk 1 above)
Comment (amitlissack) [reply api/src/opentrons/api/session.py]: Thanks. I'll add that.

**Review by mcous — APPROVED** [review]

**Pull request merged**